src/common/secret.c

```diff
@@ -107,7 +121,7 @@ int get_secret_option(const char *secret, const char *key_name,
   if (secret) {
     ret = set_kernel_secret(secret, key_name);
     if (ret < 0) {
-      if (ret == -ENODEV || ret == -ENOSYS) {
+      if (ret == -ENODEV || ret == -ENOSYS || ret == -EINVAL) {
```

Comment:
`-EINVAL` is a valid return code of `set_kernel_secret()` on GNU/Linux. if the input parameter is invalid, -EINVAL is returned. probably you can return `-ENOSYS` on FreeBSD. it's handled properly already.